Reviewing a design for a11y post (#1332)
a11yproject/a11yproject.com: src/posts/2021-08-14-reviewing-a-design-for-accessibility.md

category: Quick tip
author: Steve Barnett
date: 2021-08-14
tags:
  - quicktip
---

I'm a big believer in [shifting left](https://www.deque.com/shift-left/): considering accessibility as early in the process as we can. One way we can do this is by reviewing designs for accessibility. This helps us catch anything we've missed before we start developing it.

## What to do: annotate

By adding annotations about the semantics and accessibility of elements on a page we can remove barriers to access. The best way to do annotations might be sticky notes on a printout or [adding comments on a Figma file](https://help.figma.com/hc/en-us/articles/360041068574), or something else. The important thing is that we want to be clear and explicit.

Here are a handful of things that I've found give good value / effort balance when reviewing a design for accessibility. 

## Things to check

- **Page title**
    - Is the page title unique?
    - Does it concisely describe the page content?
- **Headings**
    - Are headings used to divide content up in a logical way?
    - Are the levels shown, without skipping any?
- **Text alternatives**

>     - Do all images (including icons) have a text alternative (using the `alt` attribute)? (See [WebAIM's guide to alternative text](https://webaim.org/techniques/alttext/) for more about what this is, and what makes good alternative text)

Reviewer: ```suggestion
    - Do all images (including icons) have a text alternative (using the `alt` attribute)? You can see [WebAIM's guide to alternative text](https://webaim.org/techniques/alttext/) for more about what this is, and what makes good alternative text.
```

I think we can promote this from a parenthetical into a full-fledged statement!

    - If the image is decorative, is it marked as such (using `alt=""`)? Are we sure it's only decorative?
- **Link and button text**
    - Does each link describe where we'll go when we follow it? Does each button describe what will happen when we use it?
    - Where links or buttons have the same text repeated on a page, do they have a longer accessible name that is unique? For example: "View details of Quarter 1 review" instead of "View details".
- **Standard controls**

>     - Do form controls match standard ones (like `selects`, `radio`, and `checkbox`) where possible? If not, is it possible to use a standard control instead? (They have accessibility baked in)

Reviewer: ```suggestion
    - Do form controls match standard ones (like `select`s, `radio`, and `checkbox`) where possible? If not, is it possible to use a standard control instead?
```

Two things here:

1. Removing "s" from the `select` element's name and moving it outside.
2. Removing the "accessibility baked in" line because if you get suuuuper technical that's not always true. I think the fact that it's posted here implies it'll be net-better for assistive technology support, though.

    - Are standard controls being used to match the design intent?
- **Names for form elements**

>     - Do all form controls (one input plus one label) have a name? (See [MDN's article on the `<label>` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for an example of how the `label` and `input` elements work together with the `id` and `for` attributes to provide accessible names)

Reviewer: ```suggestion
    - Do all form controls (one input plus one label) have a name? You can reference [MDN's article on the `<label>` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for an example of how the `label` and `input` elements work together with the `id` and `for` attributes to provide accessible names.
```

Another good sentence here. Also suggesting changing "see" to "reference" , as not everyone can see.

    - If the name isn't visible, is it shown in an annotation?
- **Names for groups of form elements**

>     - Do groups of related forms controls (such as multiple choice questions) have a name? The name should provide context for the individual questions in the group. (See [the W3C's Web Accessibility Tutorial on Grouping Controls](https://www.w3.org/WAI/tutorials/forms/grouping/) for two approaches to this)

Reviewer: ```suggestion
    - Do groups of related forms controls (such as multiple choice questions) have a name? The name should provide context for the individual questions in the group. Reference [the W3C's Web Accessibility Tutorial on Grouping Controls](https://www.w3.org/WAI/tutorials/forms/grouping/) for two approaches on how to do this.
```

Another sentence!
